WKData/Sources/WKData/Data Controllers/WKDonateDataController.swift

```diff
@@ -0,0 +1,132 @@
+        
+        // TODO: Send in API key
+        
+        var errors: [Error] = []
+        
+        group.enter()
+        let paymentMethodsRequest = WKBasicServiceRequest(url: paymentMethodsURL, method: .GET, parameters: paymentMethodParameters)
+        service.performDecodableGET(request: paymentMethodsRequest) { (result: Result<WKPaymentMethods, Error>) in
+            defer {
+                group.leave()
+            }
+            
+            switch result {
+            case .success(let paymentMethods):
+                Self.paymentMethods = paymentMethods
+            case .failure(let error):
+                errors.append(error)
+            }
+        }
+        
+        group.enter()
+        let donateConfigRequest = WKBasicServiceRequest(url: donateConfigURL, method: .GET, parameters: donateConfigParameters)
+        service.performDecodableGET(request: donateConfigRequest) { (result: Result<WKDonateConfigResponse, Error>) in
+            defer {
+                group.leave()
+            }
+            
+            switch result {
+            case .success(let response):
+                Self.donateConfig = response.config
+            case .failure(let error):
+                errors.append(error)
+            }
+        }
+        
+        group.notify(queue: .main) {
+            if let firstError = errors.first {
+                completion(.failure(firstError))
+                return
+            }
+            
+            completion(.success(()))
+        }
+    }
+    
+    public func submitPayment(amount: Decimal, currencyCode: String, paymentToken: String, donorName: String, donorEmail: String, donorAddress: String, emailOptIn: Bool?, completion: @escaping (Result<Void, Error>) -> Void) {
+        
+        guard let donatePaymentSubmissionURL = URL.donatePaymentSubmissionURL() else {
+            completion(.failure(WKDataControllerError.failureCreatingRequestURL))
+            return
+        }
+        
+        let donorInfo: [String: Any] = [
+            "name": donorName,
+            "email": donorEmail,
+            "address": donorAddress
+        ]
+        
+        var parameters: [String: Any] = [
+            "action": "submitPayment",
+            "amount": amount,
+            "currency": currencyCode,
+            "payment_token": paymentToken,
+            "donor_info": donorInfo
+        ]
+        
+        if let emailOptIn {
+            parameters["opt_in"] = emailOptIn
+        }
+        
+        // TODO: Send in API key
+            
+        let request = WKBasicServiceRequest(url: donatePaymentSubmissionURL, method: .POST, parameters: parameters, bodyContentType: .json)
+        service?.performDecodablePOST(request: request, completion: { (result: Result<WKPaymentSubmissionResponse, Error>) in
+            switch result {
+            case .success(let response):
+                guard response.response.status == "Success" else {
+                    return
```

Comment:
Missing a `completion` handler call in this failure case.

Reply:
Great catch! Done.